File: app/src/main/java/com/gooner10/popularmoviesapp/Activity/moviefavourite/MovieFavouriteContract.java
package com.gooner10.popularmoviesapp.Activity.moviefavourite;

import com.gooner10.popularmoviesapp.Activity.data.MovieData;

import java.util.List;

/**
 * Contract for the favorite fragment
 */

public class MovieFavouriteContract {
    interface View {
        void displayFavoriteMovie(List<MovieData> movieDataList);
    }

    interface Presenter {
        void loadFavoriteMovieFromRepository();
    }
}

File: feature/favoritemovie/src/main/java/com/gooner10/favorite/MovieFavouriteContract.kt
package com.gooner10.favorite

import com.gooner10.data.model.MovieItem

/**
 * Contract for the favorite fragment
 */

class MovieFavouriteContract {
    interface View {
        fun displayFavoriteMovie(movieDataList: List<MovieItem>)
    }

    internal interface Presenter {
        fun loadFavoriteMovieFromRepository()
    }
}
